sympy/physics/mechanics/functions.py

```diff
@@ -422,8 +444,17 @@ def find_dynamicsymbols(expression, exclude=None):
             raise TypeError("exclude kwarg must be iterable")
     else:
         exclude_set = set()
-    return set([i for i in expression.atoms(AppliedUndef, Derivative) if
-            i.free_symbols == t_set]) - exclude_set
+    try:
+        if isinstance(expression, Vector):
+            expr=expression.to_matrix(reference_frame)
+            return set([i for i in expr.atoms(AppliedUndef, Derivative) if
+                    i.free_symbols == t_set]) - exclude_set
+    except TypeError:
+        print(" function missing one required argument of type ReferenceFrame ")
```

Comment:
Make a complete sentence here that is more informative to the user. "You must provide a reference frame when passing in a vector expression."